### PR TITLE
[mailgun-js] add testMode and testModeLogger to Mailgun constructor for version 0.…

### PR DESCRIPTION
**Pull request by JiriBalcar**
…22.0

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/bojand/mailgun-js/blob/master/README.md#test-mode>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/mailgun-js/mailgun-js-tests.ts [hunk 1]
```diff
@@ -11,6 +13,23 @@ const mailgun2 = new mailgunFactory2({
     domain: "auth.domain"
 });
 
+const logger = (httpOptions: mailgunFactory.LoggerHttpOptions, payload: string, form: FormData) => {
+    const { method, path } = httpOptions;
+    const hasPayload = !!payload;
```
Comment (andipaetzold) [inline types/mailgun-js/mailgun-js-tests.ts]: You can remove line 18 to 21. They don't add any value to the tests.

**Review by andipaetzold — CHANGES_REQUESTED** [review]
The types look good!

Please revert the formatting changes in order to reduce the size of the diff. Looks like your editor did this automatically.
Then I am happy to approve :)

File: types/mailgun-js/index.d.ts [hunk 2]
```diff
@@ -88,7 +105,7 @@ declare namespace Mailgun {
         }
 
         interface BatchData extends SendData {
-            "recipient-variables"?: BatchSendRecipientVars;
```
Comment (JiriBalcar) [inline types/mailgun-js/index.d.ts]: changed this quotes to match previous lines

File: types/mailgun-js/mailgun-js-tests.ts (hunk 1 above)
Comment (JiriBalcar) [reply types/mailgun-js/mailgun-js-tests.ts]: removed

**Comment by JiriBalcar** [comment]
@andipaetzold I removed the formatting. Thank you for review.

**Review by andipaetzold — APPROVED** [review]

**Pull request merged**